boto/utils.py

```diff
@@ -104,6 +104,33 @@ def unquote_v(nv):
         return (nv[0], urllib.parse.unquote(nv[1]))
 
 
+def parse_qs_safe(qs, keep_blank_values=False, strict_parsing=False,
+                  encoding='utf-8', errors='replace'):
```

Comment:
What's the motivation for having `errors='replace'`?  Is there a case where a `.decode()` in `decode_qs_dict` would fail to decode?  If I change this to `strict`, the tests you've added still pass.